cannon/mipsevm/multithreaded/mips.go

```diff
@@ -206,6 +210,46 @@ func (m *InstrumentedState) handleSyscall() error {
 	return nil
 }
 
+func (m *InstrumentedState) syscallGetRandom(a0, a1 uint64) (v0, v1 uint64) {
+	// Get existing memory value at target address
+	effAddr := a0 & arch.AddressMask
+	m.memoryTracker.TrackMemAccess(effAddr)
+	memVal := m.state.Memory.GetWord(effAddr)
+
+	// Generate some pseudorandom data
+	randomWord := m.splitmix64(m.state.Step)
+
+	// Calculate number of bytes to write
+	targetByteIndex := a0 - effAddr
+	maxBytes := arch.WordSizeBytes - targetByteIndex
+	byteCount := a1
+	if maxBytes < byteCount {
+		byteCount = maxBytes
+	}
+
+	// Write random data into target memory location
+	var randDataMask arch.Word = (1 << (byteCount * 8)) - 1
+	// Shift left to align with index 0, then shift right to target correct index
+	randDataMask <<= (arch.WordSizeBytes - byteCount) * 8
+	randDataMask >>= targetByteIndex * 8
+	newMemVal := (memVal & ^randDataMask) | (randomWord & randDataMask)
+	m.state.Memory.SetWord(effAddr, newMemVal)
+
+	v0 = byteCount
+	v1 = 0
+
+	return v0, v1
+}
+
+// splitmix64 generates a pseudorandom 64-bit value.
+// See canonical implementation: https://prng.di.unimi.it/splitmix64.c
+func (m *InstrumentedState) splitmix64(seed uint64) uint64 {
```

Comment:
nit: this function doesn't depend on state:
```suggestion
func splitmix64(seed uint64) uint64 {
```